### PR TITLE
attach popup mode feature added

### PR DESCRIPTION
**Pull request by ipekbayrak**

### CODE REVIEW

File: videoengager/js/videoengager.widget.js [hunk 1]
```diff
@@ -492,15 +504,8 @@ class VideoEngager {
       document.getElementsByClassName('cx-submit')[0].innerHTML = submitButton;
     };
 
-    this.terminateInteraction = function () {
```
Comment (shadji) [inline videoengager/js/videoengager.widget.js:495]: Why this is changed? @ipekbayrak

File: videoengager/js/videoengager.widget.js [hunk 2]
```diff
@@ -440,6 +442,16 @@ class VideoEngager {
       });
     };
 
+    this.terminateWebChatInteraction = function () {
```
Comment (shadji) [inline videoengager/js/videoengager.widget.js:445]: @ipekbayrak what is the logic here>

File: videoengager/js/videoengager.widget.js (hunk 1 above)
Comment (shadji) [reply videoengager/js/videoengager.widget.js:495]: If this is used somewhere by the customers you break it,

File: videoengager/js/videoengager.widget.js [hunk 3]
```diff
@@ -630,10 +635,10 @@ class VideoEngager {
     const closeIframeOrPopup = function () {
       interactionId = null;
       if (!iframeHolder) {
-        if (popupinstance) {
+        if (popupinstance && attachPopup) {
```
Comment (shadji) [inline videoengager/js/videoengager.widget.js:638]: That;s wrong, we are told to close... @ipekbayrak

File: videoengager/js/videoengager.widget.js (hunk 1 above)
Comment (ipekbayrak) [reply videoengager/js/videoengager.widget.js:495]: fixed by having both functions

File: videoengager/js/videoengager.widget.js (hunk 3 above)
Comment (shadji) [reply videoengager/js/videoengager.widget.js:638]: we are told to close, so close, no hidden if's please

**Pull request merged**